### PR TITLE
Docs: TOC, more details, fixes

### PR DESCRIPTION
**Pull request by erikjung**
This improves the docs a bit.
- Added a table of contents (and `docs` script to use `doctoc`)
- Added more details to many sections
- Added some overview text to a few sections
- Fixed broken links

Preview: https://github.com/cloudfour/drizzle/tree/docs-wip_cont/docs

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,76 +1,313 @@
...
 
 # Getting Started
 
-Drizzle is built on the [Node.js](node) platform, so be sure to have it installed before proceeding.
+Drizzle is built on the [Node.js][node] platform, so be sure to have it installed before proceeding.
 
 ## Installation
 
-[Download](https://github.com/cloudfour/drizzle/archive/master.zip) and extract a copy of the Drizzle source, then run this command in the resulting directory:
+1. [Download][download] and extract a copy of the source.
```
Comment (saralohr) [inline docs/README.md:48]: Is the second `[download]` supposed to be a URL?

Comment (saralohr) [inline docs/README.md:48]: Nevermind! I see that this works in the link you provided.

**Comment by saralohr** [comment]
This is looking great! 👍

**Comment by gerardo-rodriguez** [comment]
Sweet! LGTM 👍

**Pull request merged**